File: packages/types/src/valuation.ts
export interface ValuationMetrics {
  symbol: string;
  name: string;
  currentPrice: number;
  peRatio: number | null;
  forwardPE: number | null;
  pegRatio: number | null;
  priceToBook: number | null;
  priceToSales: number | null;
  eps: number | null;
  bookValuePerShare: number | null;
  revenuePerShare: number | null;
  grahamNumber: number | null;
  dcfValue: number | null;
  upside: number | null;
  verdict: "undervalued" | "overvalued" | "fair" | null;
}

File: apps/mobile/lib/models/valuation.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'valuation.freezed.dart';
part 'valuation.g.dart';

@freezed
class ValuationMetrics with _$ValuationMetrics {
  const factory ValuationMetrics({
    required String symbol,
    required String name,
    required double currentPrice,
    double? peRatio,
    double? forwardPE,
    double? pegRatio,
    double? priceToBook,
    double? priceToSales,
    double? eps,
    double? bookValuePerShare,
    double? revenuePerShare,
    double? grahamNumber,
    double? dcfValue,
    double? upside,
    String? verdict,
  }) = _ValuationMetrics;

  factory ValuationMetrics.fromJson(Map<String, dynamic> json) =>
      _$ValuationMetricsFromJson(json);
}
